### PR TITLE
feat: Allow organizer to update his details (#827)

### LINKED ISSUE #827: Orga app should allow the organizer to update his/her info from the app.
**Is it a ?**

- [ ] Bug Report

- [ ] Feature Request
    
- [ ] Chore
    
    
### Feature Request

**Feature Description**

The Application does not allow the organizer to update his profile his/her information.

screenshot_2018-03-22-19-10-15

Do you want to work on this issue?
Yes

### PR DESCRIPTION
**Pull request by AkshayCHD**
Fixes #827 

Changes: Allow organizer to change his details

Screenshots for the change: 
screenshot_1523137423

3updated

### CODE REVIEW

**Comment by AkshayCHD** [comment]
@iamareebjamal please review

File: app/src/main/java/org/fossasia/openevent/app/data/event/EventApi.java [hunk 1]
```diff
@@ -30,4 +30,7 @@
 
     @GET("events/{id}/general-statistics")
     Observable<EventStatistics> getEventStatistics(@Path("id") long id);
+
+    @PATCH("users/{id}")
+    Observable<User> patchUser(@Path("id") long id, @Body User user);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/event/EventApi.java]: Add in auth API

File: app/src/main/java/org/fossasia/openevent/app/data/event/EventRepository.java [hunk 2]
```diff
@@ -20,4 +20,6 @@
     Observable<Event> createEvent(Event event);
 
     Observable<EventStatistics> getEventStatistics(long id);
+
+    Observable<User> updateOrganizer(User user);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/event/EventRepository.java]: Auth Service

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/data/event/EventRepository.java]: Change to update user

File: app/src/main/res/menu/menu_organizer.xml [hunk 3]
```diff
@@ -9,4 +9,9 @@
         android:id="@+id/action_change_password"
         android:title="@string/change_password"
         app:showAsAction="never"/>
+
+    <item android:id="@+id/update_organizer"
+        android:title="@string/sort_by"
+        android:icon="@drawable/ic_person"
```
Comment (iamareebjamal) [inline app/src/main/res/menu/menu_organizer.xml]: Icon should be of edit

File: app/src/main/res/menu/menu_organizer.xml [hunk 4]
```diff
@@ -9,4 +9,9 @@
         android:id="@+id/action_change_password"
         android:title="@string/change_password"
         app:showAsAction="never"/>
+
+    <item android:id="@+id/update_organizer"
+        android:title="@string/sort_by"
```
Comment (iamareebjamal) [inline app/src/main/res/menu/menu_organizer.xml]: Fix title

**Comment by AkshayCHD** [comment]
@iamareebjamal please review

File: app/src/main/java/org/fossasia/openevent/app/core/organizer/update/UpdateOrganizerInfoPresenter.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import org.fossasia.openevent.app.utils.StringUtils;
+
+import javax.inject.Inject;
+
+import static org.fossasia.openevent.app.common.rx.ViewTransformers.dispose;
+import static org.fossasia.openevent.app.common.rx.ViewTransformers.progressiveErroneous;
+
+public class UpdateOrganizerInfoPresenter extends AbstractBasePresenter<UpdateOrganizerInfoView> {
+
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/organizer/update/UpdateOrganizerInfoPresenter.java]: Extra line

File: app/src/main/java/org/fossasia/openevent/app/core/organizer/update/UpdateOrganizerInfoPresenter.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+            .compose(progressiveErroneous(getView()))
+            .doFinally(this::showUser)
+            .subscribe(loadedUser -> this.user = loadedUser, Logger::logError);
+    }
+
+    private void showUser() {
+        getView().setUser(user);
+    }
+
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/organizer/update/UpdateOrganizerInfoPresenter.java]: Extra line

File: app/src/main/java/org/fossasia/openevent/app/data/auth/AuthServiceImpl.java [hunk 7]
```diff
@@ -128,4 +131,41 @@ public Completable changePassword(ChangePassword changePassword) {
             .subscribeOn(Schedulers.io())
             .observeOn(AndroidSchedulers.mainThread());
     }
+
+    @Override
+    public Observable<User> getUser(boolean reload) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/auth/AuthServiceImpl.java]: This was already above as getOrganizer I guess

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -12,6 +12,17 @@
     <string name="first_name">First Name (Optional)</string>
     <string name="last_name">Last Name (Optional)</string>
     <string name="email">Email</string>
+    <string name="update_organizer_details">Update Organizer Details</string>
```
Comment (iamareebjamal) [inline app/src/main/res/values/strings.xml]: Just change to `Edit Profile`

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -12,6 +12,17 @@
...
+    <string name="update">Update</string>
+    <string name="firstname">FirstName</string>
+    <string name="lastname">LastName</string>
```
Comment (iamareebjamal) [inline app/src/main/res/values/strings.xml]: Why no spaces

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -12,6 +12,17 @@
...
+    <string name="thumbnail_image">Thumbnail Image URL</string>
+    <string name="icon_image_url">Icon Image URL</string>
```
Comment (iamareebjamal) [inline app/src/main/res/values/strings.xml]: Above two fields are read only, please remove them

File: app/src/main/java/org/fossasia/openevent/app/data/auth/AuthServiceImpl.java (hunk 7 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/data/auth/AuthServiceImpl.java]: I think getOrganizer is in EventRepository. If so, move all these methods and that to `userApi` and `userRepository`

**Comment by AkshayCHD** [comment]
@iamareebjamal so should i make a separate package for user, containing userApi, repository, model etc

**Comment by iamareebjamal** [comment]
@AkshayCHD No point of adding wrong behaviour and then correcting it

Everything in this PR is fine. Just this bit will finalize it

**Comment by iamareebjamal** [comment]
Yes you should

**Comment by AkshayCHD** [comment]
@iamareebjamal please review

**Pull request merged**